core/src/test/java/org/elasticsearch/index/seqno/LocalCheckpointServiceTests.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.seqno;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.util.concurrent.AbstractRunnable;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.IndexSettingsModule;
+import org.junit.Before;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class LocalCheckpointServiceTests extends ESTestCase {
+
+    LocalCheckpointService checkpointService;
+
+    final int SMALL_INDEX_LAG_THRESHOLD = 10;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        checkpointService = getCheckpointService(SMALL_INDEX_LAG_THRESHOLD, LocalCheckpointService.DEFAULT_INDEX_LAG_MAX_WAIT);
+    }
+
+    protected LocalCheckpointService getCheckpointService(int thresholdLag, TimeValue thresholdDelay) {
+        return new LocalCheckpointService(
+                new ShardId("test", 0),
+                IndexSettingsModule.newIndexSettings("test",
+                        Settings.builder()
+                                .put(LocalCheckpointService.SETTINGS_INDEX_LAG_THRESHOLD, thresholdLag)
+                                .put(LocalCheckpointService.SETTINGS_INDEX_LAG_MAX_WAIT, thresholdDelay)
+                                .build()
+                ));
+    }
+
+    public void testSimplePrimary() {
```

Comment:
Good, but can we also have a simple test like this one that would force the offset to wrap around?